### PR TITLE
NAS-127927 / 24.10 / Expand SMB client testing

### PR DESCRIPTION
**Pull request by anodos325**
Add explicit tests for the xattr handlers for alternate data streams.

### CODE REVIEW

**Review by bmeagherix — DISMISSED** [review]

File: tests/api2/test_425_smb_protocol.py [hunk 1]
```diff
@@ -605,6 +624,14 @@ def test_152_check_xattr_via_smb(request, xat):
     # Python base64 library appends a `\t` to end of byte string
     assert xat_bytes == bytes, str(err)
 
+    # Check via kernel client.
+    kcontent = get_stream('afp_xattr_testfile', AFPXattr[xat]['smbname'])
+    err = {
+        "name": xat,
+        "b64data": b64encode(kcontent[:-1])
+    }
+    assert xat_bytes == bytes, str(err)
```
Comment (bmeagherix) [inline tests/api2/test_425_smb_protocol.py:633]: Shouldn't this check use `kcontent` rather than `xat_bytes`?

**Review by bmeagherix — CHANGES_REQUESTED** [review]
Mostly LGTM.  One change requested.

Would also be nice to see a test with multiple streams on the same file/dir.

File: tests/api2/test_425_smb_protocol.py (hunk 1 above)
Comment (anodos325) [reply tests/api2/test_425_smb_protocol.py:633]: Yes! (doh)

File: tests/api2/test_425_smb_protocol.py [hunk 2]
```diff
@@ -275,35 +281,81 @@ def test_061_create_and_write_stream_smb2(request):
...
+        try:
+            # We have to an extra byte for that nul at end of xattr
+            offset = 2097152 - (len(b"test2") + 1)
+            c.write(fd, b"test2", offset)
+            contents = c.read(fd, offset, 5)
+            assert contents.decode() == "test2"
+        finally:
+            c.close(fd)
+
+        copy_stream('streamstestfile', 'smb2_stream', 'smb2_stream2')
```
Comment (bmeagherix) [inline tests/api2/test_425_smb_protocol.py:358]: After the `copy_stream` seems like a perfect opportunity to:
1. Run `list_stream` and ensure that the resultant set is as expected.
2. Read _smb2_stream2_ using _**both**_ SMB and `get_stream` and validate contents.

File: tests/api2/test_425_smb_protocol.py [hunk 3]
```diff
@@ -582,29 +634,37 @@ def test_151_set_xattr_via_ssh(request, xat):
...
     fd = c.create_file(afptestfile, "w")
-    xat_bytes = c.read(fd, 0, len(bytes) + 1)
+    xat_bytes = c.read(fd, 0, len(bytes_to_read) + 1)
     c.close(fd)
     c.disconnect()
 
     err = {
         "name": xat,
-        "b64data": b64encode(bytes)
+        "b64data": b64encode(bytes_to_read)
```
Comment (bmeagherix) [inline tests/api2/test_425_smb_protocol.py:654]: Shouldn't this be `xat_bytes` rather than `bytes_to_read`?

(`bytes_to_read` is the well-known input)

**Review by bmeagherix — CHANGES_REQUESTED** [review]

**Review by bmeagherix — APPROVED** [review]

**Pull request merged**